### PR TITLE
fix(legacy): add type=button to cancel buttons

### PR DESCRIPTION
**Pull request by squidsoup**
## Done

Without type="button" set, many cancel buttons are the default action when pressing 'enter' in the app, instead of submitting or saving the form.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit `/MAAS/l/networks?by=fabric`
- Select 'add fabric' from the dropdown
- Enter some text for a new fabric
- Hit enter and ensure the form as submitted, rather then cancelling

## Fixes

Fixes: #1809

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.

### CODE REVIEW

File: legacy/src/app/partials/dhcp-snippets-table.html [hunk 1]
```diff
@@ -14,11 +14,13 @@
   </thead>
```
Comment (huwshimi) [inline legacy/src/app/partials/dhcp-snippets-table.html:14]: Pretty sure this file can be deleted (it's from the old settings).

**Review by huwshimi — APPROVED** [review]
Seems OK, hard to tell with all the prettier noise.

File: legacy/src/app/partials/dhcp-snippets-table.html (hunk 1 above)
Comment (huwshimi) [reply legacy/src/app/partials/dhcp-snippets-table.html:14]: Actually, looks like this is used on node details so can't be removed just yet.

**Comment by squidsoup** [comment]
> Seems OK, hard to tell with all the prettier noise.

yeah, the prettier noise is a problem, proposing a branch shortly to prettify legacy so diffs are easier to read in the future.

**Pull request merged**